README.md
# Power-Backpacks
Factorio Mod

A mod adding backpacks as well as powered versions


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.1.0
Date: 09.06.2021
  Changes:
    - Initial Release
---------------------------------------------------------------------------------------------------

recipe.lua
-- Item Recipies
data:extend({
    {
        type = "recipe",
        name = "backpack-wooden",
        enabled = false,
        energy_required = 0.5,
        ingredients =
        {
            {"wood", 4},
        },
        result = "backpack-wooden"
    },
    {
        type = "recipe",
        name = "backpack-iron",
        enabled = false,
        energy_required = 1,
        ingredients =
        {
            {"backpack-wooden", 4},
            {"iron-plate", 16},
        },
        result = "backpack-iron"
    },
    {
        type = "recipe",
        name = "backpack-steel",
        enabled = false,
        energy_required = 1.5,
        ingredients =
        {
            {"backpack-iron", 8},
            {"steel-plate", 16},
        },
        result = "backpack-steel"
    },
})
-- Backpack Recipies
data:extend({
    { -- Light Armor
        type = "recipe",
        name = "backpack-mk1",
        enabled = false,
        order = "a[light-backpack]",
        energy_required = 3,
        ingredients =
        {
            {"iron-plate", 40},
            {"backpack-wooden", 2},
        },
        result = "backpack-mk1"
    },
    { -- Heavy Armor
        type = "recipe",
        name = "backpack-mk2",
        enabled = false,
        order = "b[heavy-backpack]",
        energy_required = 8,
        ingredients =
        {
            {"copper-plate", 100},
            {"steel-plate", 50},
            {"backpack-wooden", 8},
        },
        result = "backpack-mk2"
    },
    { -- Modular Armor
        type = "recipe",
        name = "backpack-mk3",
        enabled = false,
        order = "c[modular-backpack]",
        energy_required = 15,
        ingredients =
        {
            {"steel-plate", 50},
            {"advanced-circuit", 50},
            {"backpack-iron", 8},
        },
        result = "backpack-mk3"
    },
    { -- Power Armor
        type = "recipe",
        name = "backpack-mk4",
        enabled = false,
        order = "d[power-backpack]",
        energy_required = 20,
        ingredients =
        {
            {"steel-plate", 40},
            {"processing-unit", 40},
            {"electric-engine-unit", 20},
            {"backpack-iron", 16},
        },
        result = "backpack-mk4"
    },
    { -- Power Armor MK2
        type = "recipe",
        name = "backpack-mk5",
        enabled = false,
        order = "e[power-backpack-mk2]",
        energy_required = 30,
        ingredients =
        {
            {"processing-unit", 60},
            {"electric-engine-unit", 40},
            {"low-density-structure", 30},
            {"speed-module-2", 25},
            {"effectivity-module-2", 25},
            {"backpack-steel", 8},
        },
        result = "backpack-mk5"
    },
    { -- Power Armor MK3
        type = "recipe",
        name = "backpack-mk6",
        enabled = false,
        order = "f[power-backpack-mk3]",
        energy_required = 40,
        ingredients =
        {
            {"processing-unit", 120},
            {"electric-engine-unit", 80},
            {"low-density-structure", 80},
            {"speed-module-3", 20},
            {"effectivity-module-3", 20},
            {"backpack-steel", 16},
        },
        result = "backpack-mk6"
    },
    { -- Power Armor MK4
        type = "recipe",
        name = "backpack-mk7",
        enabled = false,
        order = "g[power-backpack-mk4]",
        energy_required = 50,
        ingredients =
        {
            {"processing-unit", 240},
            {"electric-engine-unit", 160},
            {"low-density-structure", 160},
            {"speed-module-3", 40},
            {"effectivity-module-3", 40},
            {"backpack-steel", 32},
        },
        result = "backpack-mk7"
    },
})
-- Power Armor Recipies
data:extend({
    { -- Power Armor MK3
        type = "recipe",
        name = "power-armor-mk3",
        enabled = false,
        order = "f[power-backpack-mk3]",
        energy_required = 40,
        ingredients =
        {
            {"processing-unit", 120},
            {"electric-engine-unit", 80},
            {"low-density-structure", 80},
            {"speed-module-3", 20},
            {"effectivity-module-3", 20},
        },
        result = "power-armor-mk3"
    },
    { -- Power Armor MK4
        type = "recipe",
        name = "power-armor-mk4",
        enabled = false,
        order = "g[power-backpack-mk4]",
        energy_required = 50,
        ingredients =
        {
            {"processing-unit", 240},
            {"electric-engine-unit", 160},
            {"low-density-structure", 160},
            {"speed-module-3", 40},
            {"effectivity-module-3", 40},
        },
        result = "power-armor-mk4"
    },
})

item.lua
-- Items
data:extend({
	{
		type = "item",
		name = "backpack-wooden",
		icon = "__Power-Backpacks__/graphics/backpack-wooden.png",
		icon_size = 64,
        subgroup = "raw-resource",
        --order = "a[light-armor]",
        stack_size = 64
	},
    {
		type = "item",
		name = "backpack-iron",
		icon = "__Power-Backpacks__/graphics/backpack-iron.png",
		icon_size = 64,
        subgroup = "raw-resource",
        --order = "a[light-armor]",
        stack_size = 64
	},
    
    {
		type = "item",
		name = "backpack-steel",
		icon = "__Power-Backpacks__/graphics/backpack-steel.png",
		icon_size = 64,
        subgroup = "raw-resource",
        --order = "a[light-armor]",
        stack_size = 64
	},
})
-- Backpacks
data:extend({
	{ -- Light Armor
		type = "armor",
		name = "backpack-mk1",
		icon = "__Power-Backpacks__/graphics/backpack-mk1.png",
		icon_size = 64,
		resistances =
        {
        {
            type = "physical",
            decrease = 3,
            percent = 20
        },
        {
            type = "acid",
            decrease = 0,
            percent = 20
        },
        {
            type = "explosion",
            decrease = 2,
            percent = 20
        },
        {
            type = "fire",
            decrease = 0,
            percent = 10
        }
        },
        subgroup = "armor",
        order = "a[light-backpack]",
        stack_size = 1,
        infinite = true,
        inventory_size_bonus = 20
	},
    { -- Heavy Armor
		type = "armor",
		name = "backpack-mk2",
		icon = "__Power-Backpacks__/graphics/backpack-mk2.png",
		icon_size = 64,
		resistances =
        {
        {
            type = "physical",
            decrease = 6,
            percent = 30
        },
        {
            type = "acid",
            decrease = 0,
            percent = 40
        },
        {
            type = "explosion",
            decrease = 20,
            percent = 30
        },
        {
            type = "fire",
            decrease = 0,
            percent = 30
        }
        },
        subgroup = "armor",
        order = "b[heavy-backpack]",
        stack_size = 1,
        infinite = true,
        inventory_size_bonus = 40
	},
    { -- Modular Armor
		type = "armor",
		name = "backpack-mk3",
		icon = "__Power-Backpacks__/graphics/backpack-mk3.png",
		icon_size = 64,
		resistances =
        {
        {
            type = "physical",
            decrease = 6,
            percent = 30
        },
        {
            type = "acid",
            decrease = 0,
            percent = 50
        },
        {
            type = "explosion",
            decrease = 30,
            percent = 35
        },
        {
            type = "fire",
            decrease = 0,
            percent = 40
        }
        },
        subgroup = "armor",
        order = "c[modular-backpack]",
        stack_size = 1,
        infinite = true,
        equipment_grid = "medium-equipment-grid",
        inventory_size_bonus = 60
	},
    { -- Power Armor
		type = "armor",
		name = "backpack-mk4",
		icon = "__Power-Backpacks__/graphics/backpack-mk4.png",
		icon_size = 64,
		resistances =
        {
        {
            type = "physical",
            decrease = 8,
            percent = 30
        },
        {
            type = "acid",
            decrease = 0,
            percent = 60
        },
        {
            type = "explosion",
            decrease = 40,
            percent = 40
        },
        {
            type = "fire",
            decrease = 0,
            percent = 60
        }
        },
        subgroup = "armor",
        order = "d[power-backpack]",
        stack_size = 1,
        infinite = true,
        equipment_grid = "large-equipment-grid",
        inventory_size_bonus = 80
	},
    { -- Power Armor MK2
		type = "armor",
		name = "backpack-mk5",
		icon = "__Power-Backpacks__/graphics/backpack-mk5.png",
		icon_size = 64,
		resistances =
        {
        {
            type = "physical",
            decrease = 20,
            percent = 60
        },
        {
            type = "acid",
            decrease = 20,
            percent = 60
        },
        {
            type = "explosion",
            decrease = 30,
            percent = 60
        },
        {
            type = "fire",
            decrease = 30,
            percent = 80
        }
        },
        subgroup = "armor",
        order = "e[power-backpack-mk2]",
        stack_size = 1,
        infinite = true,
        equipment_grid = "large-equipment-grid",
        inventory_size_bonus = 100
	},
    { -- Power Armor MK3
		type = "armor",
		name = "backpack-mk6",
		icon = "__Power-Backpacks__/graphics/backpack-mk6.png",
		icon_size = 64,
		resistances =
        {
        {
            type = "physical",
            decrease = 20,
            percent = 60
        },
        {
            type = "acid",
            decrease = 20,
            percent = 60
        },
        {
            type = "explosion",
            decrease = 30,
            percent = 60
        },
        {
            type = "fire",
            decrease = 30,
            percent = 80
        }
        },
        subgroup = "armor",
        order = "f[power-backpack-mk3]",
        stack_size = 1,
        infinite = true,
        equipment_grid = "large-equipment-grid",
        inventory_size_bonus = 120
	},
    { -- Power Armor MK4
		type = "armor",
		name = "backpack-mk7",
		icon = "__Power-Backpacks__/graphics/backpack-mk7.png",
		icon_size = 64,
		resistances =
        {
        {
            type = "physical",
            decrease = 30,
            percent = 60
        },
        {
            type = "acid",
            decrease = 30,
            percent = 60
        },
        {
            type = "explosion",
            decrease = 40,
            percent = 60
        },
        {
            type = "fire",
            decrease = 0,
            percent = 100
        }
        },
        subgroup = "armor",
        order = "g[power-backpack-mk4]",
        stack_size = 1,
        infinite = true,
        equipment_grid = "large-equipment-grid",
        inventory_size_bonus = 140
	},
})
-- Backpacks
data:extend({
    { -- Power Armor MK3
		type = "armor",
		name = "power-armor-mk3",
		icon = "__Power-Backpacks__/graphics/power-armor-mk3.png",
		icon_size = 64,
		resistances =
        {
        {
            type = "physical",
            decrease = 20,
            percent = 60
        },
        {
            type = "acid",
            decrease = 20,
            percent = 60
        },
        {
            type = "explosion",
            decrease = 30,
            percent = 60
        },
        {
            type = "fire",
            decrease = 30,
            percent = 80
        }
        },
        subgroup = "armor",
        order = "f[power-armor-mk3]",
        stack_size = 1,
        infinite = true,
        equipment_grid = "large-equipment-grid",
        inventory_size_bonus = 50
	},
    { -- Power Armor MK4
		type = "armor",
		name = "power-armor-mk4",
		icon = "__Power-Backpacks__/graphics/power-armor-mk4.png",
		icon_size = 64,
		resistances =
        {
        {
            type = "physical",
            decrease = 30,
            percent = 60
        },
        {
            type = "acid",
            decrease = 30,
            percent = 60
        },
        {
            type = "explosion",
            decrease = 40,
            percent = 60
        },
        {
            type = "fire",
            decrease = 0,
            percent = 100
        }
        },
        subgroup = "armor",
        order = "g[power-armor-mk4]",
        stack_size = 1,
        infinite = true,
        equipment_grid = "large-equipment-grid",
        inventory_size_bonus = 60
	},
})

-- Roboport Mark 2